jlm/rvsdg/Trace.hpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2025 Håvard Krogstie <krogstie.havard@gmail.com>
+ * See COPYING for terms of redistribution.
+ */
+
+#ifndef JLM_RVSDG_TRACE_HPP
+#define JLM_RVSDG_TRACE_HPP
+
+#include <jlm/rvsdg/node.hpp>
+
+namespace jlm::rvsdg
+{
+class GammaNode;
+class ThetaNode;
+
+/**
+ * Helper class for tracing through RVSDG graphs to find the origins of outputs.
+ * Traces through simple nodes that do not affect the value,
+ * and through structural nodes when the value is invariant.
+ * Traces out of regions
+ */
+class OutputTracer
+{
+public:
+  virtual ~OutputTracer();
+
+  /**
+   * Creates an OutputTracer with the given configuration
+   * @param isDeep if true, tracing goes into subregions, trying to trace through structural nodes.
+   *        Otherwise, outputs of structural nodes use a simple invariant check.
+   * @param isInterprocedural allow tracing out of loops
```

Comment:
Did you mean "allow tracing out of lambdas" here? Otherwise, this makes no sense to me as interprocedural does not mean "out of loops".